src/Client/ProgressTable.cpp

```diff
@@ -268,37 +287,21 @@ void ProgressTable::writeTable(
 
         message << RESET_COLOR;
         message << CLEAR_TO_END_OF_LINE;
+
+        ++row_num;
     }
 
-    message << moveUpNLines(getFreshMetricsCount(elapsed_sec));
+    if (!is_final)
+        message << CLEAR_TO_END_OF_SCREEN << moveUpNLines(1 + row_num);
+    else
+        message << "\n\n" << SHOW_CURSOR;
+
     message.next();
 }
 
-void ProgressTable::writeFinalTable()
+void ProgressTable::writeFinalTable(WriteBufferFromFileDescriptor & message, std::unique_lock<std::mutex> & lock)
 {
-    std::lock_guard lock{mutex};
-    const auto & event_name_to_event = getEventNameToEvent();
-
-    size_t terminal_width = getTerminalWidth(in_fd, err_fd);
-    if (terminal_width < column_event_name_width + COLUMN_VALUE_WIDTH)
-        return;
-
-    if (metrics.empty())
-        return;
-
-    output_stream << "\n";
-    writeWithWidth(output_stream, COLUMN_EVENT_NAME, column_event_name_width);
-    writeWithWidth(output_stream, COLUMN_VALUE, COLUMN_VALUE_WIDTH);
-
-    for (auto & [name, per_host_info] : metrics)
-    {
-        output_stream << "\n";
-        writeWithWidth(output_stream, name, column_event_name_width);
-
-        auto value = per_host_info.getSummaryValue();
-        auto value_type = getValueType(event_name_to_event.at(name));
-        writeWithWidth(output_stream, formatReadableValue(value_type, value), COLUMN_VALUE_WIDTH);
-    }
+    writeTable(message, lock, true, true, true);
```

Comment:
`/*show_table*/ true, /*toggle_enabled*/ false, /*is_final*/ true`

It's minor since the 4th parameter doesn't affect the rendering when the 3rd is `true,` but by the time the execution reaches this point, the thread responsible for toggling the flag is no more.